ethcore/src/client.rs

```diff
@@ -391,10 +390,16 @@ impl<V> Client<V> where V: Verifier {
 			};
 
 			// Commit results
+			let closed_block = closed_block.unwrap();
+			let receipts = closed_block.block().receipts().clone();
 			closed_block.drain()
 				.commit(header.number(), &header.hash(), ancient)
 				.expect("State DB commit failed.");
 
+			// And update the chain
```

Comment:
It was introduced here: https://github.com/ethcore/parity/commit/bcaed67eaa0257c8998925287fd2dbc40df12698
to avoid race conditions when chain contains uncommitted hashes. Will add a comment explaining the order.